sopel/logger.py

```diff
@@ -119,49 +119,23 @@ def setup_logging(settings):
     dictConfig(logging_config)
 
 
-@deprecated(
-    'Use `sopel.logger.get_plugin_logger` instead',
-    version='7.0',
-    removed_in='8.0')
 def get_logger(name=None):
     """Return a logger for a module, if the name is given.
 
-    This is equivalent to ``logging.getLogger('sopel.modules.' + name)`` when
-    name is given, and ``logging.getLogger('sopel')`` when it is not.
-    The latter case is intended for use in Sopel's core; modules should call
-    ``get_logger(__name__)`` to get a logger.
-
     .. deprecated:: 7.0
 
         Use ``logging.getLogger(__name__)`` in Sopel's code instead, and
-        :func:`get_plugin_logger` for external plugins.
+        :func:`sopel.plugins.get_logger` for external plugins.
+
+        This will warn a deprecation warning in Sopel 8.0 then removed in 9.0.
 
     """
-    if name:
-        return logging.getLogger('sopel.modules.' + name)
-    else:
+    if not name:
         return logging.getLogger('sopel')
 
+    parts = name.strip().split('.')
+    if len(parts) > 1 or parts[0] in ['sopel', 'sopel_modules']:
```

Comment:
Note that this will catch either `get_logger('my.custom.pymod')` as `get_logger('sopel')` and `get_logger('sopel_modules')`; but **not** `get_logger('sopelunker')`. Which is exactly what we want!